pkg/sql/opt/exec/execbuilder/testdata/scalar

```diff
@@ -1007,6 +1044,17 @@ vectorized: true
   table: pg_attribute@pg_attribute_attrelid_idx
   spans: [/t - /t]
 
+query T
+select 12345::regprocedure::string
+----
+12345
+
+query T
+select '12345'::regprocedure::string
+----
+12345
```

Comment:
The tests in this file are verifying query plans to test that we can correctly build query plans, so I don't think these two tests should live here too. Perhaps `pkg/sql/logictest/testdata/logic_test/cast` would be a better place?